ApplicationLibCode/FileInterface/RifRoffFileTools.cpp

```diff
@@ -533,6 +536,54 @@ std::pair<bool, std::map<QString, QString>> RifRoffFileTools::createInputPropert
                     return std::make_pair( false, keywordMapping );
                 }
 
+                // Create color legend
+                {
+                    const std::string codeNamesKeyword  = keyword + roff::Parser::postFixCodeNames();
+                    const std::string codeValuesKeyword = keyword + roff::Parser::postFixCodeValues();
+
+                    auto codeNamesSize  = reader.getArrayLength( codeNamesKeyword );
+                    auto codeValuesSize = reader.getArrayLength( codeValuesKeyword );
+
+                    if ( codeNamesSize != 0 && codeNamesSize == codeValuesSize )
+                    {
+                        const auto fileCodeNames  = reader.getStringArray( codeNamesKeyword );
+                        const auto fileCodeValues = reader.getIntArray( codeValuesKeyword );
+
+                        QStringList trimmedCodeNames;
+                        bool        anyValidName = false;
+                        for ( const std::string& codeName : fileCodeNames )
+                        {
+                            QString trimmedCodeName = QString::fromStdString( codeName ).trimmed();
+                            trimmedCodeNames.push_back( trimmedCodeName );
+
+                            if ( !trimmedCodeName.isEmpty() )
+                            {
+                                anyValidName = true;
+                            }
+                        }
+
+                        if ( anyValidName )
+                        {
+                            std::vector<std::pair<int, QString>> valuesAndNames;
+
+                            for ( int i = 0; i < static_cast<int>( trimmedCodeNames.size() ); i++ )
+                            {
+                                const auto& codeName = trimmedCodeNames[i];
+                                valuesAndNames.emplace_back( fileCodeValues[i], codeName );
+                            }
+
+                            RimColorLegendCollection* colorLegendCollection = RimProject::current()->colorLegendCollection;
+
+                            // Delete existing color legend, as new legend will be populated by values from file
+                            colorLegendCollection->deleteColorLegend( newResultName );
+
+                            colorLegendCollection->createColorLegend( newResultName, valuesAndNames );
```

Comment:
Is it ok to delete the color legend here? If you import two different roff only the last one will survive.

Reply:
Thanks for the input. Use `caseId` in addition to result name when matching legend and results.